core/src/main/java/org/mapfish/print/map/geotools/grid/GridUtils.java

```diff
@@ -110,14 +113,25 @@ public static String createLabel(final double value, final String unit) {
         if (Math.abs(value - Math.round(value)) < zero) {
             return String.format("%d %s", Math.round(value), unit);
         } else {
-            if (value > maxBeforeNoDecimals || value < minBeforeScientific) {
+            if (format != null) {
+                return String.format(format, value, unit);
+            } else if ("m".equals(unit)) {
+                // meter: no decimals
                 return String.format("%1.0f %s", value, unit);
-            } else if (value < maxWithDecimals) {
-                return String.format("%1.2f %s", value, unit);
-            } else if (value > minBeforeScientific) {
-                return String.format("%1.4f %s", value, unit);
+            } else if (NonSI.DEGREE_ANGLE.toString().equals(unit)) {
+                // degree: by default 6 decimals
+                return String.format("%1.6f %s", value, unit);
             } else {
-                return String.format("%e %s", value, unit);
+                // otherwise: try to find a "nice" format
+                if (value > maxBeforeNoDecimals || value < minBeforeScientific) {
+                    return String.format("%1.0f %s", value, unit);
+                } else if (value < maxWithDecimals) {
+                    return String.format("%1.2f %s", value, unit);
+                } else if (value > minBeforeScientific) {
+                    return String.format("%1.4f %s", value, unit);
+                } else {
+                    return String.format("%e %s", value, unit);
+                }
```

Comment:
Yes, I think the common case is degree or meters. Should we simply use "%f %s" as default? (I think "%f" is better than "%e" because you don't always want the scientific notation.)